code/modules/projectiles/guns/projectile/flaregun.dm

```diff
@@ -0,0 +1,47 @@
+/obj/item/gun/projectile/flare
+	name = "flaregun"
+	desc = "A single shot polymer flare gun, the XI-54 \"Sirius\" is a reliable way to launch flares away from yourself."
+	on_mob_icon = 'icons/obj/guns/flaregun.dmi'
+	icon = 'icons/obj/guns/flaregun.dmi'
+	icon_state = "world"
+	fire_sound = 'sound/weapons/empty.ogg'
+	fire_sound_text = "a satisfying 'thump'"
+	slot_flags = SLOT_BELT | SLOT_HOLSTER
+	w_class = ITEM_SIZE_SMALL
+	obj_flags = 0
+	slot_flags = SLOT_BELT | SLOT_HOLSTER
+	matter = list(MATERIAL_STEEL = 1500, MATERIAL_PLASTIC = 2000)
```

Comment:
matter is set differently nowdays, it's dynamic from material and this list.
So in your case you should set material to plastic (since it's main material here I guess) and matter to list(MATERIAL_STEEL  = MATTER_AMOUNT_REINFORCEMENT)

Reply:
Thanks, was trying to figure that one out

Reply:
So how does it figure rest of amounts anyway? Item size?

Reply:
`w_class` is used as a multiplier.